pkg/client/googlegroup_github/mapper.go

```diff
@@ -19,15 +19,88 @@ import (
 	"context"
 	"fmt"
 	"os"
+	"slices"
 	"strings"
 
 	"google.golang.org/protobuf/encoding/prototext"
 
 	"github.com/abcxyz/pkg/logging"
 	"github.com/abcxyz/team-link/apis/v1alpha3"
+	"github.com/abcxyz/team-link/pkg/github"
 	"github.com/abcxyz/team-link/pkg/groupsync"
 )
 
+// GroupMapper implements groupsync.OneToManyGroupMapper
+// For now it was be used as Mapper between GoogleGroup
+// and GitHubTeam.
+// Both GoogleGroupMapper and GitHubMapper have similar
+// variables and functions, this way we don't need to repeatly
+// write similar function for both on them.
+type GroupMapper struct {
+	mappings map[string][]string
+}
+
+func (m *GroupMapper) AllGroupIDs(ctx context.Context) ([]string, error) {
```

Comment:
Why do these need to accept a `context.Context`? (throughout)

Reply:
This is how the interface is defined. I am considering removing this if I confirm all the current implementations (like the one currently in G3) doesn't really use `ctx`.

https://github.com/abcxyz/team-link/blob/main/pkg/groupsync/groups.go#L51